### ISSUE TITLE
Unclear routing documentation

### ISSUE THREAD

**Issue by tomkuijsten** [ocpi/ocpi#277]
In this paragraph:

https://github.com/ocpi/ocpi/blob/master/transport_and_format.asciidoc#message-routing-headers

> When OCPI is used to communicate to/from a Platform or via a Hub (which is almost always the case) the following four HTTP headers are to be added to any request/response to allow messages to be routed.

Please, be clear about the "which is almost always the case". What other cases are you talking about? Peer-to-peer? Multi-peer?

> When implementing OCPI these four headers SHALL be implemented for any request/response to/from a Functional Module. This does not mean they have to be present in all request. There are situation/special request where some headers can or shall be omitted, more on this later.

Really confusing, be clear about when to use it and when not. Don't reference "more on this later", because to understand this principle I have to read the entire doc and guess what you meant with "more on this later".

The module chapters are pretty good at describing the expected url's, parameters, content etc. But not very good at which headers are supposed to be added in which topology. It would be very helpful if you could find a way to add required/optional headers for each topology in the module section.

**Comment by RobertdeLeeuw-IHomer** [comment]
The message routing section of OCPI is one of the newer parts and has not been reviewed by many before OCPI 2.2 was released. Any feedback is always welcome, it helps us improve the specification. 

We decided not to add all the routing headers to all the request in each Functional Module. We tried that, that made the modules less readable, we also have never added the authorization header and pagination headers in the response. these are defined in generic sections. We are working on OpenAPI documentation, that hopefully shows all the headers better with each request.

I have added the exception case where message routing is not needed: Peer-to-peer connection between two parties that both only have 1 party and role implemented. So for example a pure CPO to pure MSP connection

I will remove the "more on this later" and replace it with a clickable link to the correct section in the document.
